source/slang/slang-check-decl.cpp

```diff
@@ -3724,64 +3732,155 @@ namespace Slang
         visitAbstractStorageDeclCommon(decl);
     }
 
-    void SemanticsDeclHeaderVisitor::visitAccessorDecl(AccessorDecl* decl)
+    Type* SemanticsDeclHeaderVisitor::_getAccessorStorageType(AccessorDecl* decl)
     {
-        // An accessor must appear nested inside a subscript declaration (today),
-        // or a property declaration (when we add them). It will derive
-        // its return type from the outer declaration, so we handle both
-        // of these checks at the same place.
-        auto parent = decl->parentDecl;
-        if (auto parentSubscript = as<SubscriptDecl>(parent))
+        auto parentDecl = decl->parentDecl;
+        if (auto parentSubscript = as<SubscriptDecl>(parentDecl))
         {
             ensureDecl(parentSubscript, DeclCheckState::CanUseTypeOfValueDecl);
-            decl->returnType = parentSubscript->returnType;
+            return parentSubscript->returnType;
         }
-        else if (auto parentProperty = as<PropertyDecl>(parent))
+        else if (auto parentProperty = as<PropertyDecl>(parentDecl))
         {
             ensureDecl(parentProperty, DeclCheckState::CanUseTypeOfValueDecl);
-            decl->returnType = parentProperty->type;
+            return parentProperty->type.type;
+        }
+        else
+        {
+            return getASTBuilder()->getErrorType();
         }
+    }
+
+    void SemanticsDeclHeaderVisitor::_visitAccessorDeclCommon(AccessorDecl* decl)
+    {
+        // An accessor must appear nested inside a subscript or property declaration.
+        //
+        auto parentDecl = decl->parentDecl;
+        if (as<SubscriptDecl>(parentDecl))
+        {}
+        else if (as<PropertyDecl>(parentDecl))
+        {}
         else
         {
             getSink()->diagnose(decl, Diagnostics::accessorMustBeInsideSubscriptOrProperty);
         }
+    }
 
-        checkCallableDeclCommon(decl);
+    void SemanticsDeclHeaderVisitor::visitAccessorDecl(AccessorDecl* decl)
+    {
+        _visitAccessorDeclCommon(decl);
```

Comment:
A property cannot have `void` type, right?

Reply:
I'd like to start removing as many assumptions as possible about `void` being special in Slang, so that we can start treating it as an ordinary (but empty and zero-byte) type.

Is there a particular reason that concern came up here?

Reply:
That sounds reasonable to me.